contracts/OperationsProxy.sol

```diff
@@ -49,95 +61,147 @@ contract OperationsProxy {
 		operations = Operations(_operations);
 	}
 
-	function() public only_owner {
-		relay();
-	}
-
-	function send(address _to, uint _value, bytes _data) public payable only_owner {
-		// solium-disable-next-line security/no-call-value
-		require(_to.call.value(_value)(_data));
-		Sent(_to, _value, _data);
+	function()
+		public
+		onlyOwner
+	{
+		// solium-disable-next-line security/no-low-level-calls
+		require(address(operations).call(msg.data));
 	}
 
-	function setOwner(address _owner) public only_owner {
-		OwnerChanged(owner, _owner);
+	function setOwner(address _owner)
+		public
+		onlyOwner
+	{
+		emit OwnerChanged(owner, _owner);
 		owner = _owner;
 	}
 
-	function setDelegate(address _delegate, uint8 _track) public only_owner {
-		DelegateChanged(delegate[_track], _delegate, _track);
+	function setDelegate(address _delegate, uint8 _track)
+		public
+		onlyOwner
+	{
+		emit DelegateChanged(delegate[_track], _delegate, _track);
 		delegate[_track] = _delegate;
 	}
 
-	function setConfirmer(address _confirmer, uint8 _track) public only_owner {
-		ConfirmerChanged(confirmer[_track], _confirmer, _track);
+	function setConfirmer(address _confirmer, uint8 _track)
+		public
+		onlyOwner
+	{
+		emit ConfirmerChanged(confirmer[_track], _confirmer, _track);
 		confirmer[_track] = _confirmer;
 	}
 
-	function addRelease(bytes32 _release, uint32 _forkBlock, uint8 _track, uint24 _semver, bool _critical) public {
-		if (relayOrConfirm(_track))
-			AddReleaseRelayed(_track, _release);
-		else
+	function addRelease(
```

Comment:
Do you mind adding an additional modifier here to make sure it can only be called by `onlyDelegateFor(_track)`?
Currently it will fail calling `tryRelay` function, but imho it would be better to actually duplicate it here to make it easier to analyse.

Reply:
Agreed, I will also do the same thing for `addChecksum`. Maybe the optimizer even removes the redundant check.